core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -74,35 +75,35 @@ public void append(IntArrayList positions, Block source)
             return;
         }
 
-        if (source instanceof RunLengthEncodedBlock rleBlock) {
-            appendRle(rleBlock.getValue(), positions.size());
-        }
-        else if (source instanceof DictionaryBlock dictionaryBlock) {
-            ValueBlock dictionary = dictionaryBlock.getDictionary();
-            if (state == State.UNINITIALIZED) {
-                state = State.DICTIONARY;
-                this.dictionary = dictionary;
-                dictionaryIdsBuilder.appendPositions(positions, dictionaryBlock);
+        switch (source) {
+            case RunLengthEncodedBlock rleBlock -> {
```

Comment:
Actually this is called `pattern matching for switch` (https://openjdk.org/jeps/441) ;)